codex/sales/slotqueue.nim

```diff
@@ -386,25 +309,24 @@ proc clearSeenFlags*(self: SlotQueue) =
 
   trace "all 'seen' flags cleared"
 
-proc run(self: SlotQueue) {.async: (raises: []).} =
-  while self.running:
+proc runWorker(queue: SlotQueue) {.async: (raises: []).} =
+  trace "slot queue worker loop started"
+  while queue.running:
     try:
-      if self.paused:
+      if queue.paused:
         trace "Queue is paused, waiting for new slots or availabilities to be modified/added"
 
       # block until unpaused is true/fired, ie wait for queue to be unpaused
-      await self.unpaused.wait()
+      await queue.unpaused.wait()
 
-      let worker =
-        await self.workers.popFirst() # if workers saturated, wait here for new workers
-      let item = await self.queue.pop() # if queue empty, wait here for new items
+      let item = await queue.queue.pop() # if queue empty, wait here for new items
```

Comment:
Precisely why I preferred `self` over `queue` 😉 `queue.queue` is a bit cumbersome to me 🤷‍♂️ But also, it doesn't really matter at all.

Reply:
I did this refactor in a couple of steps. At first, I had `run(self: Worker, queue: Queue)`, but then it turned out that the `Worker` object was no longer needed, so I removed the `self` parameter. Now we can indeed rename `queue` to `self` again.